code/game/gamemodes/wizard/soulstone.dm

```diff
@@ -99,9 +99,12 @@
 /obj/structure/constructshell/attackby(obj/item/O, mob/user, params)
 	if(istype(O, /obj/item/device/soulstone))
 		var/obj/item/device/soulstone/SS = O
+		if(!iscultist(user) && !iswizard(user))
```

Comment:
Make it tell it's completely loyal to creator then. Just like xenobio golems - their loyality makes them useful both for regular crewmembers and antag bad guys. 

The problem of convincing the crew that the golem/construct is not a bad guy was always a thing. I don't think it needs any code solution.